### PR TITLE
fix/kitosudv copy roles fix

### PR DESCRIPTION
**Pull request by JacobBorchMadsen**
## KITOS Pull request template
The following procedure dictates the steps needed before a Pull request can be merged into master.

- [x] **Implement**: 
      *All requirements are implemented and unit tests are green*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [ ] **Green on integration**: 
      *All integration tests are green on integration*
      
- [x] **Request review**: 
      *Tag whomever you wish to review your code*
      
- [ ] **Review completed**: 
      *Reviewer ticks this box when review comments have been submitted*
      
- [ ] **Changes**: 
      *PR owner and reviewer agrees on which changes must be made and the changes are committed.*
      
- [ ] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [ ] **Green on integration**: 
      *All integration tests are green on integration*

### CODE REVIEW

**Review by strongmindsnan — APPROVED** [review]

File: Core.ApplicationServices/Users/Write/UserWriteService.cs [hunk 1]
```diff
@@ -139,10 +139,47 @@ public Maybe<OperationError> CopyUserRights(Guid organizationUuid, Guid fromUser
...
 
-            return _userService.GetUserInOrganization(organizationUuid, toUserUuid)
+            using var transactionManager = _transactionManager.Begin();
+            var copyResult = _userService.GetUserInOrganization(organizationUuid, toUserUuid)
                 .Bind(CanModifyUser)
-                .Match(toUser => _userRightsService.CopyRights(fromUser.Value.Id, toUser.Id, org.Value.Id, parameters), 
+                .Match(toUser =>
+                    {
+                        return FilterOutExistingRights(toUser, org.Value, parameters)
+                            .Match(
```
Comment (a-naskret) [inline Core.ApplicationServices/Users/Write/UserWriteService.cs:148]: You can use Bind here (then you won't have to provide the error)

File: Core.ApplicationServices/Users/Write/UserWriteService.cs [hunk 2]
```diff
@@ -139,10 +139,47 @@ public Maybe<OperationError> CopyUserRights(Guid organizationUuid, Guid fromUser
                 return fromUser.Error;
             }
 
-            return _userService.GetUserInOrganization(organizationUuid, toUserUuid)
+            using var transactionManager = _transactionManager.Begin();
+            var copyResult = _userService.GetUserInOrganization(organizationUuid, toUserUuid)
                 .Bind(CanModifyUser)
-                .Match(toUser => _userRightsService.CopyRights(fromUser.Value.Id, toUser.Id, org.Value.Id, parameters), 
+                .Match(toUser =>
```
Comment (a-naskret) [inline Core.ApplicationServices/Users/Write/UserWriteService.cs:145]: This one also can be a Bind

**Review by a-naskret — APPROVED** [review]
Looks good

File: Core.ApplicationServices/Users/Write/UserWriteService.cs (hunk 1 above)
Comment (JacobBorchMadsen) [reply Core.ApplicationServices/Users/Write/UserWriteService.cs:148]: The compiler doesn't like it as it's a maybe, and not a result

File: Core.ApplicationServices/Users/Write/UserWriteService.cs (hunk 2 above)
Comment (JacobBorchMadsen) [reply Core.ApplicationServices/Users/Write/UserWriteService.cs:145]: The compiler doesn't like it as it's a maybe, and not a result

**Pull request merged**